### PR TITLE
feat(admin-api): new gocloak client, updated user usecase

### PR DESCRIPTION
**Pull request by alexrodfe**
## Description

- Added new keycloak client
- Updated user usecase to use new keycloak client
- Added tests for usecase
- Updated graphql methods
- Deprecated previous access level related operations

## Motivation and Context

As now user management is maintained by keycloak, user related operations from api should be managed through a keycloak client.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [X] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: engine/admin-api/domain/entity/user_activity.go [hunk 1]
```diff
@@ -7,32 +7,24 @@ import (
...
-	UserActivityTypeStopVersion                UserActivityType = "STOP_VERSION"
-	UserActivityTypeUpdateSetting              UserActivityType = "UPDATE_SETTING"
-	UserActivityTypeUpdateVersionConfiguration UserActivityType = "UPDATE_VERSION_CONFIGURATION"
-	UserActivityTypeCreateUser                 UserActivityType = "CREATE_USER"
-	UserActivityTypeRemoveUsers                UserActivityType = "REMOVE_USERS"
-	UserActivityTypeUpdateAccessLevels         UserActivityType = "UPDATE_ACCESS_LEVELS"
-	UserActivityTypeRevokeSessions             UserActivityType = "REVOKE_SESSIONS"
-	UserActivityTypeGenerateAPIToken           UserActivityType = "GENERATE_API_TOKEN"
-	UserActivityTypeDeleteAPIToken             UserActivityType = "DELETE_API_TOKEN"
+	UserActivityTypeCreateRuntime            UserActivityType = "CREATE_RUNTIME"
```
Comment (dfinteligenz) [inline engine/admin-api/domain/entity/user_activity.go:10]: This should be changed to:

```suggestion
	UserActivityTypeCreateProduct            UserActivityType = "CREATE_PRODUCT"
```

**Review by dfinteligenz — CHANGES_REQUESTED** [review]

File: engine/admin-api/domain/usecase/user.go [hunk 2]
```diff
@@ -1,26 +1,116 @@
...
-// UserInteractor contains app logic to handle User entities.
+const (
+	getUserByIDWrapper                  = "get user by id"
+	updateUserProductPermissionsWrapper = "update user product permissions"
+	revokeUserProductPermissionsWrapper = "revoke user product permissions"
+	updateUserProductPermissionsLog     = "Updated user %q permissions for product %q: %v"
+	revokeUserProductPermissionsLog     = "Revoked user %q permissions for product %q"
+)
+
+// UserInteractor talks to the gocloak service, thus providing user realted operations.
```
[the remark this answers is not in the document]
Comment (alexrodfe) [reply engine/admin-api/domain/usecase/user.go:18]: Agreed, I'll add an interface for the user usecase

File: engine/admin-api/adapter/gql/http_handler.go [hunk 3]
```diff
@@ -17,7 +17,7 @@ import (
 func NewHTTPHandler(
 	logger logging.Logger,
 	runtimeInteractor *usecase.RuntimeInteractor,
-	userInteractor *usecase.UserInteractor,
+	userInteractor usecase.UserInteractorUsecase,
```
Comment (alexrodfe) [inline engine/admin-api/adapter/gql/http_handler.go:20]: change to struct pointer

File: engine/admin-api/adapter/gql/resolver.go [hunk 4]
```diff
@@ -175,6 +175,28 @@ func (r *mutationResolver) UpdateVersionUserConfiguration(ctx context.Context, i
...
+}
+
+func (r *mutationResolver) RevokeUserProductPermissions(ctx context.Context, input RevokeUserProductPermissionsInput) (bool, error) {
+	loggedUserID := ctx.Value("userID").(string)
+
+	err := r.userInteractor.RevokeUserProductPermissions(loggedUserID, input.TargetID, input.Product, *input.Comment)
+	if err != nil {
+		return false, err
+	}
+
```
Comment (alexrodfe) [inline engine/admin-api/adapter/gql/resolver.go:196]: return get by id

File: engine/admin-api/domain/usecase/user_test.go [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+	testUserData := suite.GetTestUserData()
+
+	suite.mockGokeycloak.EXPECT().GetUserByID(testUserData.ID).Times(1).Return(testUserData, nil)
+
+	userData, err := suite.userManager.GetUserByID(testUserData.ID)
+	suite.NoError(err)
+	suite.Equal(testUserData, userData)
+}
+
+func (suite *ContextUserManagerSuite) TestGetUserByIDErrorInGocloak() {
```
Comment (JYisus) [inline engine/admin-api/domain/usecase/user_test.go:60]: avoid using gocloak in usecase tests names

**Comment by JYisus** [comment]
We should decide if we want to use `product permissions` or `product grants` as the naming @alexrodfe @dfinteligenz

**Review by JYisus — CHANGES_REQUESTED** [review]

File: engine/admin-api/domain/usecase/user_test.go [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+	suite.mockGokeycloak.EXPECT().UpdateUserProductPermissions(testUserData.ID, testProduct, permissions).Times(1).Return(nil)
+	suite.mockUserActivityInteractor.EXPECT().RegisterUpdateProductPermissions(
+		triggerUserID,
+		testUserData.ID,
+		testProduct,
+		permissions,
+		testComment,
+	).Times(1).Return(nil)
+	suite.mockLogger.EXPECT().Infof(updateUserProductPermissionsLog, testUserData.ID, testProduct, permissions).Times(1)
```
[the remark this answers is not in the document]
Comment (alexrodfe) [reply engine/admin-api/domain/usecase/user_test.go]: Great tip! I'll add it to the tests

File: engine/admin-api/domain/usecase/user_test.go (hunk 5 above)
Comment (alexrodfe) [reply engine/admin-api/domain/usecase/user_test.go]: agreed, now that the service interface is called UserRegistry I'll use that as naming

**Review by dfinteligenz — APPROVED** [review]
Great job!

**Pull request merged**